hbase-server/src/main/java/org/apache/hadoop/hbase/master/MasterServices.java

```diff
@@ -354,6 +353,13 @@ long splitRegion(
    */
   boolean isInMaintenanceMode();
 
+  /**
+   * Checks master state before initiating action over region topology.
+   * @param action the name of the action under consideration, for logging.
+   * @return {@code true} when the caller should exit early, {@code false} otherwise.
+   */
+  boolean skipRegionManagementAction(final String action);
```

Comment:
I don't love exposing this up here on MasterServices, but imo, it's better to have the interface here than to expose its innards.

Suggestions welcome.

Reply:
> What is this? Is it substantial enough to be added to this Interface?

@saintstack I asked myself the very same question. It's this or have a `MasterServices` instance and cast it to `HMaster` because we need something kind-of secret.

Reply:
ok